tests/test_apiclient.py

```diff
@@ -1031,6 +1031,20 @@ def test_set_framework_result(self, data_client, rmock):
             'updated_by': 'user'
         }
 
+    def test_set_supplier_framework_prefill_declaration(self, data_client, rmock):
+        rmock.post(
+            "http://baseurl/suppliers/8765/frameworks/breeches",
+            json={"frameworkInterest": {"prefillDeclarationFromFrameworkSlug": "pyjamas"}},
+            status_code=200)
+
+        result = data_client.set_supplier_framework_prefill_declaration(8765, 'breeches', "pyjamas", "user")
+        assert result == {"frameworkInterest": {"prefillDeclarationFromFrameworkSlug": "pyjamas"}}
+        assert rmock.called
+        assert rmock.request_history[0].json() == {
+            'frameworkInterest': {"prefillDeclarationFromFrameworkSlug": "pyjamas"},
```

Comment:
Personally I like to use double quotes everywhere inside dictionaries like this - makes them easier to copy-paste as JSON if needs be.